spring/spring4/spring4-remote/pom.xml

```diff
@@ -22,6 +22,12 @@
         <dependency>
             <groupId>org.springframework</groupId>
             <artifactId>spring-context</artifactId>
+            <scope>provided</scope>
```

Comment:
Yes, and this is proper behavior...

Note that you have number of options deploying Spring apps, including:
* Using Spring Boot, which configures and add dependencies for you
* Configure everything manually (add `spring-beans`, `spring-context` etc)
* Using DropWizard spring integration
* Plus all variations between using Tomcat, Wildfly, Tomcat Embedded and Jetty

It is very likely that our clients/customer will want to add Infinispan to their existing application, so they already which configuration they want to use and we can't (or shouldn't) force them to use dependencies which we provide. Also, it's worth to mention that some configuration bits are different for all those scenarios. The only negative impact this change has is very small demo apps, like (probably) your example... You need to pick the right approach manually rather than rely on Infinispan to do it for you.

So after some time of maintaining Spring integration bits I started to think about Spring as an API. Implementations (Spring Boot, Drop Wizzard or even manual configuration) is something that depends on users preference.